src/UI/Implementation/Component/Button/Renderer.php

```diff
@@ -79,6 +90,38 @@ protected function maybeRenderId(Component\Component $component, $tpl) {
 		}
 	}
 
+	protected function renderMonth(Component\Button\Month $component, RendererInterface $default_renderer) {
+		$def = $component->getDefault();
+
+		for ($i = 1; $i<=12; $i++)
+		{
+			$this->toJS(array("month_".str_pad($i, 2, "0", STR_PAD_LEFT)."_long"));
+		}
+
+		$tpl = $this->getTemplate("tpl.month.html", true, true);
+
+		$month = explode("-", $def);
+		$tpl->setVariable("DEFAULT_LABEL", $this->txt("month_".str_pad($month[0], 2, "0", STR_PAD_LEFT)."_long")." ".$month[1]);
+
+		include_once("./Services/Calendar/classes/class.ilCalendarUtil.php");
+		\ilCalendarUtil::initDateTimePicker();
```

Comment:
> What would we win? We still have the dependency to Services/Calendar and now added redundancy, since two php code files would load the same js files.

The win would be, that we do not have the dependency to Service/Calendar except for the location of the JS files (which I would recommend to move as well). This might seem a bit religiously strict but we really want to keep this code here clean of such dependencies on PHP-Level.

> Even if we move it, there still should be some php wrapper being responsible to handle it, right? 

At this time, no. I would just add the two lines directly and leave the Service/Calendar as is. We can think about that as soon as we decide to move the input element that is connected to Service/Calendar as well.

>  If we want to get rid of the Services/Calendar dependency, we would need to refactor this completely and create a dependency from the old form implementation to the new ui framework instead.

That would be drastic and not at all realistic. leave Services/Calendar as is for now (except maybe move out the js library to libs).

> I think this will be much easier as soon as we have a form implementation in the new UI framework. 

I would even say, at this point such a thing would be undoable. We need the Input proposal first.

> I did not expect that this would be necessary at the current state.

No such thing is necessary, just add:
```
$tpl->addJavaScript("./libs/composer/vendor/moment/moment/min/moment-with-locales.min.js");		
$tpl->addJavaScript("./maybe_libs/bootstrap3_datepicker/bootstrap-datetimepicker.min.js");
```		
to the renderer.